engine/src/main/scala/cromwell/engine/io/nio/NioFlow.scala

```diff
@@ -152,21 +191,21 @@ class NioFlow(parallelism: Int,
 
   private def createDirectories(path: Path) = path.parent.createDirectories()
 
-  private def getFileHashForDrsPath(drsPath: DrsPath): IO[String] = {
-    val drsFileSystemProvider = drsPath.drsPath.getFileSystem.provider.asInstanceOf[DrsCloudNioFileSystemProvider]
+  /**
+    * Lazy evaluation of a Try in a delayed IO. This avoids accidentally eagerly evaluating the Try.
+    *
+    * IMPORTANT: Use this instead of IO.fromTry to make sure the Try will be reevaluated if the
+    * IoCommand is retried.
+    */
+  private def delayedIoFromTry[A](t: => Try[A]): IO[A] = IO[A] { t.get }
 
-    //Since this does not actually do any IO work it is not wrapped in IO
```

Comment:
I intentionally removed this comment. This _does_ actually do IO work because it calls Martha. Also, since this function doesn't deal with `IO` anyway, this comment didn't really make sense anymore.